ingress/multi-cluster/mci-https-backend/README.md

```diff
@@ -0,0 +1,385 @@
+
+```yaml
+apiVersion: networking.gke.io/v1
+kind: MultiClusterIngress
+metadata:
+  name: foobar-ingress
+  namespace: multi-cluster-demo
+  annotations:
+    networking.gke.io/static-ip: x.x.x.x
+    networking.gke.io/pre-shared-certs: "mci-certs"
+spec:
+  template:
+    spec:
+      backend:
+        serviceName: default-backend
+        servicePort: 443
+      rules:
+      - host: foo.$DOMAIN
+        http:
+          paths:
+            - backend:
+                serviceName: foo
+                servicePort: 443
+      - host: bar.$DOMAIN
+        http:
+          paths:
+            - backend:
+                serviceName: bar
+                servicePort: 443
+```
+
+Similar to the Kubernetes Service, the MultiClusterService (MCS) describes label selectors and other backend parameters to group pods in the desired way. This `foo` MCS specifies that all Pods with the following characteristics will be selected as backends for  `foo`:
+
+- Pods with the label `app: foo`
+- In the `multi-cluster-demo` Namespace
+- In any of the clusters that are registered as members to the Hub
+
+If more clusters are added to the Hub, then any Pods in those clusters that match these characteristics will also be registered as backends to `foo`.
+
+The MCS below also defines via annotations:
+
+- A [Backend Health Check](https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-features#direct_health) through backendconfig CRD
+- Backend protocol as [HTTPS](https://cloud.google.com/kubernetes-engine/docs/how-to/multi-cluster-ingress#application_protocols)
+
+```yaml
+apiVersion: networking.gke.io/v1
+kind: MultiClusterService
+metadata:
+  name: foo
+  namespace: multi-cluster-demo
+  annotations:
+    beta.cloud.google.com/backend-config: '{"ports": {"443":"backend-health-check"}}'
+    networking.gke.io/app-protocols: '{"https":"HTTPS"}'
+spec:
+  template:
+    spec:
+      selector:
+        app: foo
+      ports:
+      - name: https
+        protocol: TCP
+        port: 443
+        targetPort: 443
+```
+
+Each of the three MCS's referenced in the `foobar-ingress` MCI have their own manifest to describe the matching parameters of that MCS. A BackendConfig resource is also referenced. This allows settings specific to a Service to be configured. We use it here to configure the health check that the Google Cloud load balancer uses.
+
+```yaml
+apiVersion: cloud.google.com/v1
+kind: BackendConfig
+metadata:
+  name: backend-health-check
+  namespace: multi-cluster-demo
+spec:
+  healthCheck:
+    requestPath: /healthz
+    port: 443
+    type: HTTPS
+```
+
+Now that you have the background knowledge and understanding of MCI, you can try it out yourself.
+
+## Try it out
+
+1. Download this repo and navigate to this folder
+
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/gke-networking-recipes.git
+    Cloning into 'gke-networking-recipes'...
+
+    cd gke-networking-recipes/ingress/multi-cluster-ingress/mci-https-backend
```

Comment:
Should it be gke-networking-recipes/ingress/multi-cluster/mci-https-backend ?

Reply:
Fruits of auto-complete. Fixed it now.